vmdb/app/models/ems_kubernetes.rb

```diff
@@ -13,17 +13,41 @@ def self.description
 
   def self.raw_connect(hostname, port, api_version)
     require 'kubeclient'
-    require 'uri'
-
-    uri = URI::HTTP.build(:path => "/api", :port => port.to_i)
+    api_endpoint = raw_api_endpoint(hostname, port)
+    Kubeclient::Client.new(api_endpoint, api_version)
+  end
 
+  def self.raw_api_endpoint(hostname, port)
+    require 'uri'
     # URI::Generic#hostname= was added in ruby 1.9.3 and will automatically
     # wrap an ipv6 address in []
-    uri.hostname = hostname
-    Kubeclient::Client.new uri.to_s, api_version
+    URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/api')
```

Comment:
You did.   I just didn't see your message before I replied
:stuck_out_tongue:
 On Feb 20, 2015 2:28 PM, "Joe Rafaniello" <notifications@github.com> wrote:

> In vmdb/app/models/ems_kubernetes.rb
> <https://github.com/ManageIQ/manageiq/pull/1679#discussion_r25095744>:
>
> >      # URI::Generic#hostname= was added in ruby 1.9.3 and will automatically
> >      # wrap an ipv6 address in []
> > -    uri.hostname = hostname
> > -    Kubeclient::Client.new uri.to_s, api_version
> > +    URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/api')
>
> Yeah, that's what I said... I thought.
>
> You have to create a URI object and call .hostname=.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/manageiq/pull/1679/files#r25095744>.
>